src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.FruitStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private FruitStrategy fruitStrategy;
+
+    public ShopServiceImpl(FruitStrategy fruitStrategy) {
+        this.fruitStrategy = fruitStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactionList) {
+        transactionList
+                .forEach(transaction -> fruitStrategy.get(transaction.getOperation())
+                        .performOperation(transaction));
+    }
+}

```

Comment:
It's good to see that you have made your ShopService implementation independent and injected the FruitStrategy through the constructor, adhering to the SOLID principles.